File: VKAHUB/frontend/src/types/certificate.ts
export interface Certificate {
  id: number;
  user_id: number;
  title: string;
  description?: string;
  category?: string;
  date?: string;
  file_url: string;
  created_at: string;
  updated_at: string;
}

export interface CreateCertificateData {
  title: string;
  description?: string;
  category?: string;
  date?: string;
  file_url: string;
}

export interface UpdateCertificateData {
  title?: string;
  description?: string;
  category?: string;
  date?: string;
  file_url?: string;
}

File: VKAHUB/backend/app/domain/entities/certificate.py
"""Certificate domain entity"""

from dataclasses import dataclass
from datetime import datetime, date
from typing import Optional


@dataclass
class CertificateEntity:
    """Certificate domain entity"""

    id: int
    user_id: int
    title: str
    description: Optional[str]
    category: Optional[str]
    date: Optional[date]
    file_url: str
    created_at: datetime
    updated_at: datetime
